fix: json encode serialize atrpcs

JeremyTubongbanua opened the pull request:
closes #1765 

**- What I did**

- Changed `.toJson().toString()` to `jsonEncode()`

**- How to verify it**

See AtRpc Logs

The response String has `"` quotations so now it is JSON parsable

```
INFO|2025-12-22 14:16:39.763431| AtRpcClient |Got ack : {"reqId":1766430998869659,"respType":"ack","payload":{},"m
essage":null} 
INFO|2025-12-22 14:16:39.818674| AtRpcClient |Got response {success: true, message: 1 clients found., value: [{id:
 1, name: Jeremy, atSign: @jeremy}]} 
INFO|2025-12-22 14:16:39.818857| AtRpcClient |Got success response : {"reqId":1766430998869659,"respType":"success
","payload":{"success":true,"message":"1 clients found.","value":[{"id":"1","name":"Jeremy","atSign":"@jeremy"}]},
"message":"1 clients found."} 
```

**- Description for the changelog**
fix: json encode serialize atrpcs

JeremyTubongbanua committed:
fix: json encode serialize atrpcs

gkc reviewed (changes requested):
lgtm but two things
- can you add a unit test for this please?
- update changelog and pubspec (and AtClientConfig)

JeremyTubongbanua committed:
feat: at_rpc_types_test.dart

JeremyTubongbanua committed:
docs: update at_client/CHANGELOG.md

JeremyTubongbanua committed:
chore: format strings to be < 80 line length

JeremyTubongbanua committed:
chore: dart format at_client

JeremyTubongbanua committed:
fix: add space

JeremyTubongbanua committed:
feat: test for JSON parsability

JeremyTubongbanua committed:
chore: change it to use jsonDecode

JeremyTubongbanua committed:
chore: dart format

JeremyTubongbanua committed:
Merge branch 'trunk' into jt-atrpc-string-quotes

JeremyTubongbanua commented:
> lgtm but two things
> 
> * can you add a unit test for this please?

I've added unit tests which tests `toString` and `toJson` has JSON parsable outputs

> * update changelog and pubspec (and AtClientConfig)

I updated CHANGELOG

I noticed pub.dev's version is 3.9.2 and current trunk is 3.10.0, so I don't think this warrants a version bump to 3.11.0 or anything right?

gkc committed:
Merge branch 'trunk' into jt-atrpc-string-quotes

gkc commented:
> I noticed pub.dev's version is 3.9.2 and current trunk is 3.10.0, so I don't think this warrants a version bump to 3.11.0 or anything right?

LGTM
